pkg/ipamd/ipamd.go

```diff
@@ -884,41 +884,37 @@ func (c *IPAMContext) tryAllocateENI(ctx context.Context) error {
 		subnet = eniCfg.Subnet
 	}
 
-	eni, err := c.awsClient.AllocENI(c.useCustomNetworking, securityGroups, subnet)
-	if err != nil {
-		log.Errorf("Failed to increase pool size due to not able to allocate ENI %v", err)
-		ipamdErrInc("increaseIPPoolAllocENI")
-		return err
-	}
-
 	resourcesToAllocate := c.GetENIResourcesToAllocate()
-	_, err = c.awsClient.AllocIPAddresses(eni, resourcesToAllocate)
-	if err != nil {
-		log.Warnf("Failed to allocate %d IP addresses on an ENI: %v", resourcesToAllocate, err)
-		// Continue to process the allocated IP addresses
-		ipamdErrInc("increaseIPPoolAllocIPAddressesFailed")
-		if containsInsufficientCIDRsOrSubnetIPs(err) {
-			log.Errorf("Unable to attach IPs/Prefixes for the ENI, subnet doesn't seem to have enough IPs/Prefixes. Consider using new subnet or carve a reserved range using create-subnet-cidr-reservation")
-			c.lastInsufficientCidrError = time.Now()
+	if resourcesToAllocate > 0 {
```

Comment:
TBH we don't need this check because the func is `tryAllocateENI` and we come here only if we need to increase the DS pool..Maybe for now we can leave it but we should revisit it later. Also can you add a debug line if we ever land here and resourceToAllocate is not greater than 0...